python -m pytest aaa_doctest_2.py
============================================================================= test session starts ==============================================================================
platform win32 -- Python 3.9.7, pytest-7.2.0, pluggy-1.0.0
rootdir: C:\Users\Дмитрий
collected 3 items                                                                                                                                                                

aaa_doctest_2.py ..F                                                                                                                                                      [100%]

=================================================================================== FAILURES =================================================================================== 
___________________________________________________________________________ test_eval[..........-S] ____________________________________________________________________________ 

test_input = '..........', expected = 'S'

    @pytest.mark.parametrize("test_input,expected", [('... --- ...', 'SOS'), ('.- -...', 'AB'), ('..........', 'S')])


    def test_eval(test_input, expected):
>       assert decode(test_input) == expected

aaa_doctest_2.py:44:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _  
aaa_doctest_2.py:33: in decode
    decoded_letters = [
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _  

.0 = <list_iterator object at 0x000001FCDD5BF910>

    decoded_letters = [
>       MORSE_TO_LETTER[letter] for letter in morse_message.split()
    ]
E   KeyError: '..........'

aaa_doctest_2.py:34: KeyError
=========================================================================== short test summary info ============================================================================ 
FAILED aaa_doctest_2.py::test_eval[..........-S] - KeyError: '..........'
========================================================================= 1 failed, 2 passed in 0.12s ==========================================================================